optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,512 @@
+"""
+
+INPUTS_DOCSTRING = r"""
+    Args:
+        input_ids (`torch.Tensor`):
+            Indices of input sequence tokens in the vocabulary.
+            Indices can be obtained using [`AutoTokenizer`](https://huggingface.co/docs/transformers/autoclass_tutorial#autotokenizer).
+            [What are input IDs?](https://huggingface.co/docs/transformers/glossary#input-ids)
+        attention_mask (`torch.Tensor`), *optional*):
+            Mask to avoid performing attention on padding token indices. Mask values selected in `[0, 1]`:
+            - 1 for tokens that are **not masked**,
+            - 0 for tokens that are **masked**.
+            [What are attention masks?](https://huggingface.co/docs/transformers/glossary#attention-mask)
+        token_type_ids (`torch.Tensor`, *optional*):
+            Segment token indices to indicate first and second portions of the inputs. Indices are selected in `[0, 1]`:
+            - 1 for tokens that are **sentence A**,
+            - 0 for tokens that are **sentence B**.
+            [What are token type IDs?](https://huggingface.co/docs/transformers/glossary#token-type-ids)
+"""
+
+IMAGE_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.Tensor`):
+            Pixel values corresponding to the images in the current batch.
+            Pixel values can be obtained from encoded images using [`AutoFeatureExtractor`](https://huggingface.co/docs/transformers/autoclass_tutorial#autofeatureextractor).
+"""
+
+
+@add_start_docstrings(
+    """
+    Base INCBaseModel class.
+    """,
+)
+class INCBaseModel(OptimizedModel):
+    _AUTOMODELS_TO_TASKS = {cls_name: task for task, cls_name in TasksManager._TASKS_TO_AUTOMODELS.items()}
+    base_model_prefix = "inc_model"
+    auto_model_class = AutoModel
+    export_feature = None
+
+    def __init__(
+        self,
+        model,
+        config: PretrainedConfig = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        **kwargs,
+    ):
+        self.model = model
+        self.config = config
+        self.model_save_dir = model_save_dir
+        self.preprocessors = kwargs.get("preprocessors", [])
+        self._device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self.backend = getattr(config, "backend", None)
+
+        if is_transformers_version("<=", "4.25.1"):
+            self.generation_config = None
+        else:
+            from transformers import GenerationConfig
+
+            self.generation_config = GenerationConfig.from_model_config(config) if self.can_generate() else None
+
+        # Avoid warnings when creating a transformers pipeline
+        AutoConfig.register(self.base_model_prefix, AutoConfig)
+        self.auto_model_class.register(AutoConfig, self.__class__)
+
+    @staticmethod
+    def load_model(file_name: Union[str, Path]):
```

Comment:
Many methods are shared with `TSModelForCausalLM` would it be possible to do a refactorization in order to avoid redundancy?

Reply:
We could create a parent class `BaseModelForGeneration` in `generation/modeling.py` for both `TSModelForCausalLM` and `INCModelForCausalLM` (and later `INCModelForSeq2SeqLM`)